src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -320,9 +386,80 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
     }
   }
 
+  def groupEvents(eventType: EventType, clusteredBeamMeta: ClusteredBeamMeta): DateTime = {
+    allowGranularityChange = false
+    val eventTimestamp = timestamper(eventType)
+    val eventBucket = tuning.segmentBucket(eventTimestamp)
+    // ZK does not know about any beams
+    if (clusteredBeamMeta.equals(ClusteredBeamMeta.empty) || clusteredBeamMeta.beamDictss.isEmpty) {
+      eventBucket.start
+    } else {
+      val latestBeam = clusteredBeamMeta.beamDictss.get(clusteredBeamMeta.beamDictss.map(_._1).max).get.head
+      val latestBeamInterval = new Interval(latestBeam("interval"))
+      val newSegGranMillis = tuning.segmentGranularity.bucket(new DateTime()).toInterval.toDurationMillis
```

Comment:
Does this value change?  It seems pretty static, if so can we compute it once and store instead of computing every time?